src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -5815,25 +5815,24 @@ MergeTreeData::MutableDataPartPtr MergeTreeData::cloneAndLoadDataPartOnSameDisk(
     bool copy_instead_of_hardlink)
 {
     /// Check that the storage policy contains the disk where the src_part is located.
-    bool does_storage_policy_allow_same_disk = false;
+    bool is_on_same_disk = false;
     for (const DiskPtr & disk : getStoragePolicy()->getDisks())
     {
         if (disk->getName() == src_part->volume->getDisk()->getName())
         {
-            does_storage_policy_allow_same_disk = true;
+            is_on_same_disk = true;
             break;
         }
     }
-    if (!does_storage_policy_allow_same_disk)
-        throw Exception(
-            "Could not clone and load part " + quoteString(src_part->getFullPath()) + " because disk does not belong to storage policy",
-            ErrorCodes::BAD_ARGUMENTS);
+
+    auto target_disk = is_on_same_disk ? src_part->volume->getDisk() : getStoragePolicy()->getRandomDisk();
```

Comment:
Why do you choose to get random disk? What about taking first one to have expected behavior?

Reply:
I used getAnyDisk() at first time,  but in our cluster, we have mutiple hdd disks in a single machine, in order to improve total IO performance, we try to make data more even, so I choose random strategy here. maybe we can put a setting  here, let user to decide which strategy they want.

Reply:
Yes, setting is ok, I suppose